make/dev.mk

```
@@ -27,3 +29,12 @@ endif
 	kubectl set image deployment cluster-api-runtime-extensions-nutanix webhook=ko.local/cluster-api-runtime-extensions-nutanix:$$(gojq -r .version dist/metadata.json)
 	kubectl rollout restart deployment cluster-api-runtime-extensions-nutanix
 	kubectl rollout status deployment cluster-api-runtime-extensions-nutanix
+
+
+.PHONY: dev.update-aws-bootstrap-credentials
+dev.update-aws-bootstrap-credentials: export KUBECONFIG := $(KIND_KUBECONFIG)
+dev.update-aws-bootstrap-credentials: export AWS_B64ENCODED_CREDENTIALS := $(shell clusterawsadm bootstrap credentials encode-as-profile)
+dev.update-aws-bootstrap-credentials:
+	kubectl patch secret capa-manager-bootstrap-credentials -n capa-system -p='{"data":{"credentials": "$(AWS_B64ENCODED_CREDENTIALS)"}}'
+	kubectl rollout restart deployment capa-controller-manager -n capa-system
+	kubectl rollout status deployment capa-controller-manager -n capa-system
```

Putting this as a `shell` call makes this always run, regardless of make target requested, which isn't great. Let's move into the recipe:

```suggestion
dev.update-aws-bootstrap-credentials:
	    kubectl patch secret capa-manager-bootstrap-credentials -n capa-system -p="{\"data\":{\"credentials\": \"$$(clusterawsadm bootstrap credentials encode-as-profile)\"}}"
	kubectl rollout restart deployment capa-controller-manager -n capa-system
	kubectl rollout status deployment capa-controller-manager -n capa-system
```

---

Woah even if its part of a target? TIL

---

Yep check it out: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/8668361364/job/23773281909?pr=507#step:5:9

This is one of those make quirks that makes me want to move away from it.

---

Looking at https://www.gnu.org/software/make/manual/html_node/Reading-Makefiles.html seems like you could use a similar `=` instead of `:=` to defer assignment. These kind of make semantics (50 ways to ~leave your lover~ declare variables) are so tricky.

---

thanks. I didnt realized this. Let me test and fix it.

---

fixed: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/507/commits/8b6de5062577c1cf5a6c0e891c85a783dc03f1f3